### DISCUSSION TITLE
Why are 2D textures sometimes invisible from certain perspectives

### DISCUSSION THREAD

**Discussion by Haidiye00** [google/filament#7251] [Q&A] [answered]
this is the video:
![20231010120545_rec_](https://github.com/google/filament/assets/132528822/53d96a98-9074-4edc-95cd-cf8a22040868)
this is the Coordinate values of 2D textures:
![MA0coz4xpg](https://github.com/google/filament/assets/132528822/f555ba8a-53eb-4c2c-b6d5-f52bab7a6af8)
can you help me?

**Comment by romainguy** [comment]
How do you render that object? Is it double sided? Do you have depth testing on?

**Comment by Haidiye00** [comment]
![N0lIYQD40E](https://github.com/google/filament/assets/132528822/c228567d-bfc3-4765-bbc3-3c017e3bcb24)
this is the material

**Reply by romainguy** [reply]
Instead of using z=-1f use z=-0.9999f.

**Reply by romainguy** [reply]
Also why are you doing 2 texture fetches? Just assign the result of texture() to baseColor.

**Reply by Haidiye00** [reply]
how to fix?

**Reply by Haidiye00** [reply]
![g1lERSx4NI](https://github.com/google/filament/assets/132528822/c77ebf3e-f86f-4758-8074-5425b9113cf6)
got it,but the problem still exists

**Comment by Haidiye00** [comment]
![N0lIYQD40E](https://github.com/google/filament/assets/132528822/bea10779-ddd9-451c-b144-91c0b0efba4d)
setDepthWrite and set z=-0.9999f.Unable to solve the problem

**Reply by romainguy** [reply]
Try 0.9999f, I always forget which way our Z is pointing.

**Reply by Haidiye00** [reply]
0.9999f is skybox's depth, The following effect is not what I expected
![DJ36phcUFj](https://github.com/google/filament/assets/132528822/86c991a6-dc7d-4aef-a2d2-7a2362472379)

**Reply by romainguy** [reply]
Yeah just checking the coordinates where in the right way.

**Reply by Haidiye00** [reply]
Isn't the text supposed to be at the front of the screen,？it shouldn't be obstructed

**Reply by romainguy** [reply]
Yeah, -0.9999 is the right value to use. Make sure to disable depth culling and culling though.

**Reply by Haidiye00** [reply]
I hope the text rendering can always be at the forefront of the screen

**Reply by Haidiye00** [reply, marked as answer]
![LTd96eoPZu](https://github.com/google/filament/assets/132528822/96987713-f9a4-4609-ac67-924aa6a544e0)
culling = false ,I found the problem solved
